src/git-auth-helper.ts

```diff
@@ -150,24 +151,60 @@ class GitAuthHelper {
 
   async configureSubmoduleAuth(): Promise<void> {
     // Remove possible previous HTTPS instead of SSH
-    await this.removeGitConfig(this.insteadOfKey, true)
+    await this.removeSubmoduleGitConfig(this.insteadOfKey)
 
     if (this.settings.persistCredentials) {
-      // Configure a placeholder value. This approach avoids the credential being captured
```

Comment:
The code to get the submodule git-config file paths moved into `this.git.getSubmoduleConfigPaths(...)`